src/core/fetchers/segment/prioritizer.ts

```diff
@@ -136,87 +223,236 @@ export default class ObservablePrioritizer<T> {
    * @param {number} priority
    * @returns {Observable}
    */
-  public create(obs : Observable<T>, priority : number) : Observable<T> {
+  public create(obs : Observable<T>, priority : number) : Observable<ITaskEvent<T>> {
     const pObs$ = observableDefer(() => {
-      if (this._pendingPriority == null || this._pendingPriority >= priority) {
-        // Update the priority and start immediately the Observable
-        this._pendingPriority = priority;
-        return this._startObservable(obs);
-      } else {
-        const trigger = new Subject<void>();
-        this._queue.push({ observable: pObs$, priority, trigger });
-        return trigger
-          .pipe(mergeMap(() => this._startObservable(obs)));
+      const trigger = new Subject<boolean>();
+
+      /**
+       * Function executed each time the `trigger` Subject emits.
+       * @param {Boolean} shouldRun - If `true`, the observable can run. If
+       * `false` it means that it has just been interrupted.
+       * @returns {Observable} - Returns events corresponding to the lifecycle
+       * of the observable.
+       */
+      const onTrigger = (shouldRun : boolean) : Observable<ITaskEvent<T>> => {
+        if (!shouldRun) {
+          return observableOf({ type: "interrupted" as const });
+        }
+        this._onTaskBegin(newTask);
+        return observableConcat(
+          obs.pipe(map((data) => ({ type: "data" as const, value: data }))),
+          observableOf({ type: "ended" as const })
+        ).pipe(finalize(() => {
+          this._onTaskEnd(newTask);
+        }));
+      };
+
+      const newTask : IPrioritizerTask<T> = { observable: pObs$, priority, trigger };
```

Comment:
It may be clearer to declare the newTask before declaring onTrigger.

Reply:
Ok, will move

Reply:
Done